Modules/_sre/sre.h

```diff
@@ -18,10 +18,10 @@
 #define SRE_CODE Py_UCS4
 #if SIZEOF_SIZE_T > 4
 # define SRE_MAXREPEAT (~(SRE_CODE)0)
-# define SRE_MAXGROUPS ((~(SRE_CODE)0) / 2)
+# define SRE_MAXGROUPS ((SRE_CODE)INT32_MAX / 2)
 #else
 # define SRE_MAXREPEAT ((SRE_CODE)PY_SSIZE_T_MAX)
-# define SRE_MAXGROUPS ((SRE_CODE)PY_SSIZE_T_MAX / SIZEOF_SIZE_T / 2)
+# define SRE_MAXGROUPS ((SRE_CODE)INT32_MAX / SIZEOF_INT / 2)
```

Comment:
It should be `SIZEOF_VOID_P` in the denominator (it is the size of the `mark` item) and `PY_SSIZE_T_MAX` in the numerator, because the condition is no integer overflow in `state->mark = PyMem_New(const void *, pattern->groups * 2)`.

Reply:
done